### PR TITLE
WAITP-1411: Setup datatrak survey screens

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Update routes with survey routes
- Survey screen components

---

### Screenshots:
image

### CODE REVIEW

File: packages/datatrak-web/src/api/queries/useProjects.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useQuery } from 'react-query';
+
+export const useProjects = () => {
```
Comment (tcaiger) [inline packages/datatrak-web/src/api/queries/useProjects.tsx:8]: This is just placeholder

File: packages/datatrak-web/src/api/queries/useSurveys.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useQuery } from 'react-query';
+
+export const useSurveys = () => {
```
Comment (tcaiger) [inline packages/datatrak-web/src/api/queries/useSurveys.tsx:8]: This is just placeholder

File: packages/datatrak-web/src/views/NotFoundPage.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+
+export const NotFoundPage = () => {
```
Comment (tcaiger) [inline packages/datatrak-web/src/views/NotFoundPage.tsx:7]: This is just placeholder

File: packages/datatrak-web/src/views/SurveyReviewPage.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+const Container = styled.div`
+  margin-top: 3rem;
+  margin-bottom: 3rem;
+  padding: 3rem;
+  background: white;
+  max-width: 60rem;
+  border-radius: 3px;
+  border: 1px solid #dfdfdf;
+`;
+export const SurveyReviewPage = () => {
```
Comment (tcaiger) [inline packages/datatrak-web/src/views/SurveyReviewPage.tsx:21]: This is just placeholder

File: packages/datatrak-web/src/views/SurveySuccessPage.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { Typography } from '@material-ui/core';
+
+const Container = styled.div`
```
Comment (tcaiger) [inline packages/datatrak-web/src/views/SurveySuccessPage.tsx:10]: This is just placeholder

File: packages/datatrak-web/src/views/SurveyPage.tsx [hunk 6]
```diff
@@ -3,7 +3,47 @@
...
+import { SurveyParams } from '../types';
+import { HEADER_HEIGHT } from '../constants';
+
+const Container = styled.div`
+  height: calc(100vh - ${HEADER_HEIGHT});
+  display: flex;
+  flex-direction: column;
+`;
+
+const Toolbar = styled.div`
```
Comment (tcaiger) [inline packages/datatrak-web/src/views/SurveyPage.tsx:20]: This is just placeholder

File: packages/datatrak-web/src/features/Survey/SurveyScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                  questionOptions,
+                  config,
+                  questionName,
+                  questionLabel,
+                  validationCriteria,
+                },
+                index,
+              ) => {
+                if (validationCriteria?.mandatory === true) {
+                  console.log('mandatory question', questionCode);
```
Comment (acdunham) [inline packages/datatrak-web/src/features/Survey/SurveyScreen.tsx:134]: I'm guessing this is placeholder until we know what the validation will do?

**Review by acdunham — APPROVED** [review]
This looks great. I like the use of the context here. Just one small request to add in the types to the query hook for survey components, but otherwise approving

**Pull request merged**